src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartRpc.cs

```diff
@@ -30,6 +32,11 @@ public StartRpc(INethermindApi api)
         public async Task Execute(CancellationToken cancellationToken)
         {
             IJsonRpcConfig jsonRpcConfig = _api.Config<IJsonRpcConfig>();
+            IKeyStoreConfig keyStoreConfig = _api.Config<IKeyStoreConfig>();
+
+            // updating jwt-secret path based on data directory.
+            jsonRpcConfig.JwtSecretFile = String.Concat(keyStoreConfig.KeyStoreDirectory,"/jwt-secret");
```

Comment:
This change actually introduces another bug disrespecting the `JsonRpc.JwtSecretFile` setting. I think we should remove the default value of the `IJsonRpcConfig.JwtSecretFile` and check for null. If null, then apply this change.